imap_processing/codice/codice_l1a.py

```diff
@@ -290,6 +291,119 @@ def unpack_science_data(self, science_values: str) -> None:
         self.data = [["1"] * 128] * self.num_counters
 
 
+def create_event_dataset(
+    met: list[int], event_data: str, dataset_name: str, data_version: str
+) -> xr.Dataset:
+    """
+    Create dataset for event data.
+
+    Parameters
+    ----------
+    met : list[int]
+        The Mission Elapsed Time of the data.
+    event_data : str
+        A string of binary numbers representing the event data.
+    dataset_name : str
+        The name for the dataset.
+    data_version : str
+        Version of the data product being created.
+
+    Returns
+    -------
+    dataset : xarray.Dataset
+        Xarray dataset containing the event data.
+    """
+    cdf_attrs = ImapCdfAttributes()
+    cdf_attrs.add_instrument_global_attrs("codice")
+    cdf_attrs.add_instrument_variable_attrs("codice", "l1a")
+    cdf_attrs.add_global_attribute("Data_version", data_version)
+
+    # Define coordinates
+    epoch = xr.DataArray(
+        met_to_j2000ns(met),  # TODO: Fix after SIT-3 (see note below)
+        name="epoch",
+        dims=["epoch"],
+        attrs=cdf_attrs.get_variable_attributes("epoch_attrs"),
+    )
+
+    # Create the dataset to hold the data variables
+    dataset = xr.Dataset(
+        coords={
+            "epoch": epoch,
+        },
+        attrs=cdf_attrs.get_global_attributes(dataset_name),
+    )
+
+    # TODO: Determine what should go in event data CDF and how it should be
+    # structured.
+
+    return dataset
+
+
+def create_hskp_dataset(
```

Comment:
This function used to be in `codice.utils`, but since I am defining the `create_event_dataset` and `create_science_dataset` functions in this module, I figured it would be better if `create_hskp_dataset` was in here too.